src/snowflake/cli/plugins/nativeapp/commands.py

```diff
@@ -285,3 +285,17 @@ def app_deploy(
     return MessageResult(
         f"Deployed successfully. Application package and stage are up-to-date."
     )
+
+
+@app.command("validate", requires_connection=True)
+@with_project_definition("native_app")
+def app_validate(**options):
```

Comment:
A few thoughts on the expected behaviour for this:
- do we target the current local version of the setup script, prior to full deployment, or the version that's already deployed? As I argued offline, I think we offer more value if it's the former. We could also let the user choose, but we'd have to think of a good mechanism for that.
- what happens when the user specifies `--format json` for the `validate` command? I can see tools being built on this functionality.
- should we validate as part of other commands too, e.g. `snow app deploy` or `run`? I'm not sure if GS runs validation by default ATM, but it would good for the user to see the output in other flows.

Reply:
1 and 3 are addressed in the design doc. For 2, I'm thinking of just returning the raw response and skip the whole abort behaviour. Do we support running `snow app deploy` or `snow app run` with `--format json` too?

Reply:
All commands support it, but run and deploy don't really output anything worth including in JSON. Validate is a little different.